### PR TITLE
Feature/integrate react routing

### PR DESCRIPTION
**Pull request by nathf**

### CODE REVIEW

File: app/main/views/suppliers.py [hunk 1]
```diff
@@ -202,50 +242,84 @@ def edit_supplier_case_study(casestudy_id):
     )
 
 
-def _validate_case_study(casestudy):
+def _validate_case_study(casestudy, extra_fields=[]):
```
Comment (matt-sm) [inline app/main/views/suppliers.py]: I just added https://github.com/AusDTO/dto-digitalmarketplace-utils/blob/master/react/response.py#L14 refactor to use this

**Comment by matt-sm** [comment]
how does this all work?

File: app/main/views/suppliers.py [hunk 2]
```diff
@@ -88,15 +88,20 @@ def get_supplier_case_study(casestudy_id):
...
 @login_required
 def new_supplier_case_study():
     if not current_user.role == 'supplier':
         flash('buyer-role-required', 'error')
         return current_app.login_manager.unauthorized()
     form = DmForm()
+    basename = url_for('.new_supplier_case_study')
     rendered_component = render_component('bundles/CaseStudy/CaseStudyWidget.js', {
-        'form_options': {'csrf_token': form.csrf_token.current_token},
-        'casestudy': {'returnLink': url_for('main.get_supplier', code=current_user.supplier_code)}})
```
Comment (matt-sm) [inline app/main/views/suppliers.py]: should returnLink still be there?

File: app/main/views/suppliers.py [hunk 3]
```diff
@@ -202,36 +254,43 @@ def edit_supplier_case_study(casestudy_id):
...
-def update_supplier_case_study(casestudy_id):
+def update_supplier_case_study(casestudy_id, step=None):
     old_casestudy = DataAPIClient().get_case_study(casestudy_id)['caseStudy']
     supplier_code = old_casestudy.get('supplierCode') if old_casestudy else None
 
     if not can_view_supplier_page(supplier_code):
         flash('buyer-role-required', 'error')
         return current_app.login_manager.unauthorized()
 
+    extra_fields = []
```
Comment (maxious) [inline app/main/views/suppliers.py]: is this used anywhere?

**Comment by matt-sm** [comment]
- so we edit across 2 pages, but view on one page?
- when I click case study, update, add text in reference, click save - I end up on a 404 at http://localhost:8000/reference
- should the reference url be something like /case-study/id/update/reference rather than /reference?
- click edit, then click reference, then click case study (on right menu) I'm now at '/' - shouldn't that be /case-study/id/update/?
- i thought reference was optional?  seems i need to enter all those values
- when i add new case study, get all validation on page 1 working, click save, on page 2 i'm presented with validation straight away - shouldnt this validation appear on publish click?

**Pull request merged**